Refactor settings modules

sergei-maertens opened the pull request:
This should bring some order in the chaos that were the settings modules.

* Pull as much config out of the environment as possible
* Restructure environment-specific settings modules
* Refactor environment specific modules to have as little overrides as
  possible

sergei-maertens commented:
Obviously need to fix some tests :-)

joeribekker commented on src/openzaak/conf/base.py:
```diff
@@ -332,17 +417,25 @@
 if SENTRY_DSN:
     INSTALLED_APPS = INSTALLED_APPS + ["raven.contrib.django.raven_compat"]
 
-    if "GIT_SHA" in os.environ:
```
This should return. Steven moved it here because it could crash if no SENTRY/DSN was present.

sergei-maertens replied:
Fixed in f91534797675e1bd4d5febc321fa048a57866edd

joeribekker commented on src/openzaak/conf/local_example.py:
```diff
@@ -1,25 +1,13 @@
-#
-# Any machine specific settings when using development settings.
-#
+from .dev import DATABASES, INSTALLED_APPS
```
```suggestion
from .envs.dev import DATABASES, INSTALLED_APPS
```

sergei-maertens replied:
Should be fixed by changing.envs into .includes and swapping them

joeribekker commented on .travis.yml:
```diff
@@ -18,8 +18,7 @@ addons:
   global:
-  - DJANGO_SETTINGS_MODULE=openzaak.conf.test
-  - SECRET_KEY=dummy
+  - DJANGO_SETTINGS_MODULE=openzaak.conf.envs.ci
```
Can you please reverse this? Add config "parts" in a package "includes" and keep the config targets in `conf`.

sergei-maertens replied:
Yep :+1:

joeribekker commented on docs/conf.md:
```diff
@@ -0,0 +1,116 @@
+* start the Open Zaak processes (with uwsgi/gunicorn/celery) in a process
+  manager that defines the environment variables
+
+### Providing a .env file
```
Can you add `.env-example` as well to the branch? It should have all these constants imo.

sergei-maertens replied:
Sure thing

sergei-maertens commented:
@joeribekker 

* fixed docker build
* fixed tests running on Travis
* fixed missing admin template
* swapped the `.envs` for `.includes`
* added `.env-example` file
